x-pack/plugins/maps/public/routes/map_page/map_app/map_app.tsx

```diff
@@ -50,13 +51,17 @@ import {
   unsavedChangesWarning,
 } from '../saved_map';
 import { waitUntilTimeLayersLoad$ } from './wait_until_time_layers_load';
+import { SpacesPluginStart } from '../../../../../spaces/public';
+import { resolveSavedObject } from '../../../resolve_saved_object';
 
 interface MapRefreshConfig {
   isPaused: boolean;
   interval: number;
 }
 
 export interface Props {
+  spacesApi?: SpacesPluginStart;
+  http: HttpStart;
```

Comment:
http does not need to be passed to MapApp, its never used. This dependency chain starts all the way in RenderApp. No need to pass http to RenderApp in public/plugin